$OpenBSD: patch-src_thread_c,v 1.7 2010/12/31 12:03:55 dcoppa Exp $

fix an assertion failure

fix from upstream: set the MEDIAENDED state correctly when running
as plugin

--- src/thread.c.orig	Fri Nov  5 17:16:22 2010
+++ src/thread.c	Fri Dec 24 14:13:53 2010
@@ -448,7 +448,7 @@ gboolean thread_reader(GIOChannel * source, GIOConditi
 		if (embed_window == 0)
 			idledata->cachepercent = 0.0;
 
-		if (g_ascii_strncasecmp(vo, "gl2", strlen("gl2")) != 0) {
+		if (vo != NULL && g_ascii_strncasecmp(vo, "gl2", strlen("gl2")) != 0) {
 		    cmd = g_strdup_printf("brightness %i\n", idledata->brightness);
 		    send_command(cmd, TRUE);
 		    g_free(cmd);
@@ -1495,6 +1495,7 @@ gpointer launch_player(gpointer data)
 
                     // nothing is on the playlist and we are not looping so ask plugin for next item
                     if (embed_window != 0 || control_id != 0) {
+                        js_state = STATE_MEDIAENDED;
                         dbus_send_event("MediaComplete", 0);
                         dbus_open_next();
                     }
